src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -94,14 +94,20 @@ public override void OnEventWritten(string name, object payload)
                         _ = this.connectionFetcher.TryFetch(command, out var connection);
                         _ = this.databaseFetcher.TryFetch(connection, out var database);
 
-                        activity.DisplayName = (string)database;
+                        if (database != null)
+                        {
+                            activity.DisplayName = (string)database;
+                        }
 
                         _ = this.dataSourceFetcher.TryFetch(connection, out var dataSource);
                         _ = this.commandTextFetcher.TryFetch(command, out var commandText);
 
-                        activity.SetTag(SemanticConventions.AttributeDbName, (string)database);
+                        activity.SetTag(SemanticConventions.AttributeDbName, database);
```

Comment:
Not directly related to this PR.

Shouldn't it be in databased != statement? I think that we do not need empty/null attribute,

Reply:
good catch, I missed that.
I'm fine making this change in this PR, but let me know if this should be done in a follow up instead.

Reply:
I would put it in scope of this pr